normalizer/core.py

```diff
@@ -53,53 +63,154 @@ def order_py_modules(py_modules: List['pathlib.Path'], work_path: 'pathlib.Path'
     return orders
 
 
-def inspect_executors(py_modules: List['pathlib.Path']):
+def _get_element_source(
+        lines: List[str],
+        element: ast.expr,
+        remove_whitespace: bool = False) -> str:
+    if element.lineno == element.end_lineno:
+        source = lines[element.lineno - 1][element.col_offset:element.end_col_offset]
+    else:
+        source = lines[element.lineno - 1][element.col_offset:] + \
+                     ''.join(lines[element.lineno:element.end_lineno - 1]) + \
+                     lines[element.end_lineno - 1][:element.end_col_offset]
+    if remove_whitespace:
+        source = re.sub(r'\s+', '', source)
+        source = source.replace(',', ', ')
+    return source
+
+
+def _get_args_kwargs(
+        func_args: List[str],
+        func_args_defaults: List[str],
+        annotations: List[Optional[str]]) -> Tuple[ArgType, KWArgType]:
+    if len(func_args_defaults) == 0:
+        kwargs_idx = len(func_args)
+    else:
+        kwargs_idx = -len(func_args_defaults)
+    kwarg_arguments, kwargs_annotations, kwargs_defaults = func_args[kwargs_idx:],\
+                                                           annotations[kwargs_idx:],\
+                                                           func_args_defaults
+
+    kwargs = [
+        (arg, annotation, default)
+        for arg, annotation, default in zip(kwarg_arguments, kwargs_annotations, kwargs_defaults)
+    ]
+
+    arg_arguments, args_annotations = func_args[:kwargs_idx], annotations[:kwargs_idx]
+    args = [
+        (arg, annotation)
+        for arg, annotation in zip(arg_arguments, args_annotations)
+    ]
+    return args, kwargs
+
+
+def _inspect_requests(element: ast.FunctionDef, lines: List[str]) -> Optional[str]:
+    """
+    Returns requests inspection details about a method
+    This can return:
+        * None : the method is not an endpoint (not decorated with @requests)
+        * "'ALL'": this is an endpoint method that will be triggered on every endpoint (decorated with @requests)
+        * "'/<endpoint>'": this is an endpoint method that will be triggered on '/<endpoint>' (decorated with
+        @requests(on='/<endpoint>')
+        * "['/<endpoint1>', '/<endpoint2>', ...]": this is an endpoint method that will be triggered on any of the
+        endpoints in the list
```

Comment:
So, in this case, the function would return different results for `@requests(on='/endpoint')`and `@requests(on=['/endpoint'])`? Would it be better to make the results returned for two cases identical? And is there a reason to return this as a list converted string (e.g. `"['/endpoint1', '/endpoint2']"` as opposed to an actual list (e.g. `['/endpoint1', '/endpoint2']`)?

Reply:
@winstonww  I just pushed a commit that makes  `@requests(on='/endpoint')` and `@requests(on=['/endpoint'])` return the same value (`"['/endpoint']"`)
I was discussion with miao the fact that we can make this return a list instead of string, however, this is not valid, because, we don't know which value to set for `ALL`. We can't use `[]` because `@requests(on=[])` means nothing will trigger the endpoint

Reply:
Gotcha